test/store_unittest.cc

```diff
@@ -0,0 +1,234 @@
+#include "../src/configuration.h"
+#include "../src/resource.h"
+#include "../src/store.h"
+#include "../src/time.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataStoreTest : public ::testing::Test {
+ protected:
+  Configuration config;
+  MetadataStore store;
+
+  MetadataStoreTest() : config(), store(config) {}
+
+  std::map<MonitoredResource, MetadataStore::Metadata> GetMetadataMap() const {
+    return store.GetMetadataMap();
+  }
+
+  void PurgeDeletedEntries() {
+    store.PurgeDeletedEntries();
+  }
+
+};
+
+TEST_F(MetadataStoreTest, ResourceWithOneIdStoredCorrectly) {
+  MonitoredResource mr("some_resource", {});
+  store.UpdateResource({"id"}, mr);
+  EXPECT_EQ(mr, store.LookupResource("id"));
+}
+
+TEST_F(MetadataStoreTest, EmptyStoreThrowsError) {
+  EXPECT_THROW(store.LookupResource("some_resource_id"), std::out_of_range);
+}
+
+TEST_F(MetadataStoreTest, ResourceNotFoundThrowsError) {
+  MonitoredResource mr("some_resource", {});
+  store.UpdateResource({"id"}, mr);
+  EXPECT_EQ(mr, store.LookupResource("id"));
+  EXPECT_THROW(store.LookupResource("other_resource"), std::out_of_range);
+}
+
+TEST_F(MetadataStoreTest, MultipleResourcesWithSingleIdsCorrectlyStored) {
+  MonitoredResource mr1("some_resource1", {});
+  MonitoredResource mr2("some_resource2", {});
+  store.UpdateResource({"id1"}, mr1);
+  store.UpdateResource({"id2"}, mr2);
+  EXPECT_EQ(mr1, store.LookupResource("id1"));
+  EXPECT_EQ(mr2, store.LookupResource("id2"));
```

Comment:
[Optional] Switch the order of lookups around, just to make it more interesting? Also below.

Reply:
Done.